src/internal/packager/template/template.go

```diff
@@ -91,9 +91,10 @@ func (values Values) Apply(component types.ZarfComponent, path string) {
 	}
 
 	// Include the data injection marker template if the component has data injections
-	// TODO: (@jeff-mccoy) DATA_INJECTON_MARKER is misspelled however it is in use by other packages and would be a breaking change!
 	if len(component.DataInjections) > 0 {
+		// Preserve existing misspelling for backwards compatibility
 		builtinMap["DATA_INJECTON_MARKER"] = config.GetDataInjectionMarker()
```

Comment:
Actually, can we track this into templating and warn the user that they should update their spelling?  (assuming we would want to remove this one day).

Reply:
It's a string replacement, little worried about breaking something if we do that, ideas?

Reply:
We could have a deprecated list of keys that it would search for first and just display a warning about them.  We could even have that be a string to string map so that we could loop through the keys to search and then display a message like:

`Warning:  This Zarf Package uses a deprecated variable: "DATA_INJECTON_MARKER" changed to "DATA_INJECTION_MARKER".  Please notify your package creator for an update.`

Reply:
@Madeline-UX on potential wording there

Reply:
so this is a string replace, so this would be adding an additional scan to the files?

Reply:
That would be the easiest way to do it, though ideally only for the limited subset of deprecations.  We are already scanning the file bytes many times over.

Reply:
```
// ReplaceTextTemplate loads a file from a given path, replaces text in it and writes it back in place.
func ReplaceTextTemplate(path string, mappings map[string]string) {
	text, err := os.ReadFile(path)
	if err != nil {
		message.Fatalf(err, "Unable to load %s", path)
	}

	for deprecated, value := range template.Deprecations {
		if bytes.Contains(text, deprecated) {
			message.Warnf("This Zarf Package uses a deprecated variable: '%s' changed to '%s'.  Please notify your package creator for an update.", &deprecated, &value)
		}
	}

	for template, value := range mappings {
		text = bytes.ReplaceAll(text, []byte(template), []byte(value))
	}

	if err = os.WriteFile(path, text, 0600); err != nil {
		message.Fatalf(err, "Unable to update %s", path)
	}
}
```